lib/api/readable.js

```diff
@@ -0,0 +1,272 @@
+// Ported from https://github.com/nodejs/undici/pull/907
+
+'use strict'
+
+const assert = require('assert')
+const { Readable } = require('stream')
+const { RequestAbortedError, NotSupportedError } = require('../core/errors')
+
+let Blob
+
+const kConsume = Symbol('kConsume')
+const kReading = Symbol('kReading')
+const kBody = Symbol('kBody')
+const kAbort = Symbol('abort')
+
+module.exports = class BodyReadable extends Readable {
```

Comment:
I didn't get a chance to read through the old PR, but why not use the experimental WebStreams stuff? https://nodejs.org/api/webstreams.html#webstreams_class_readablestream

My personal plan for undici-fetch was to build on top of that

Reply:
> I didn't get a chance to read through the old PR, but why not use the experimental WebStreams stuff?

Performance and ergonomics. Also making sure it doesn't break the existing API.

Reply:
Also I guess it makes it easier for upstream libraries to also implement fetch body mixin.

Reply:
Very smart! LGTM